docs/source/get_started/minimal_kedro_project.md

```diff
@@ -0,0 +1,176 @@
+
+Typically, it looks similar to this:
+```toml
+[tool.kedro]
+package_name = "package_name"
+project_name = "project_name"
+kedro_init_version = "kedro_version"
+tools = ""
+example_pipeline = "False"
+source_dir = "src"
+```
+
+This informs Kedro where to look for the source code, `settings.py` and `pipeline_registry.py` are.
+
+#### `settings.py`
+The `settings.py` file is an important configuration file in a Kedro project that allows you to define various settings and hooks for your project. Here’s a breakdown of its purpose and functionality:
+- Project Settings: This file is where you can configure project-wide settings, such as defining the logging level, setting environment variables, or specifying paths for data and outputs.
+- Hooks Registration: You can register custom hooks in `settings.py`, which are functions that can be executed at specific points in the Kedro pipeline lifecycle (e.g., before or after a node runs). This is useful for adding additional functionality, such as logging or monitoring.
+- Integration with Plugins: If you are using Kedro plugins, `settings.py` can also be utilized to configure them appropriately.
+
+Even if you do not have any settings, an empty `settings.py` is still required. Typically, they are stored at `src/<package_name>/settings.py`.
+
+#### `pipeline_registry.py`
+The `pipeline_registry.py` file is essential for managing the pipelines within your Kedro project. It provides a centralized way to register and access all pipelines defined in the project. Here are its key features:
+- Pipeline Registration: The file must contain a top-level function called `register_pipelines()` that returns a mapping from pipeline names to Pipeline objects. This function is crucial because it enables the Kedro CLI and other tools to discover and run the defined pipelines.
+- Autodiscovery of Pipelines: Since Kedro 0.18.3, you can use the [`find_pipeline`](../nodes_and_pipelines/pipeline_registry.md#pipeline-autodiscovery) function to automatically discover pipelines defined in your project without manually updating the registry each time you create a new pipeline.
+
+## Creating a Minimal Kedro Project Step-by-Step
+This guide will walk you through the process of creating a minimal Kedro project, allowing you to successfully run `kedro run` with just three files.
+
+### Step 1: Install Kedro
+
+First, ensure that Python is installed on your machine. Then, install Kedro using pip:
+
+```bash
+pip install kedro
+```
+
+### Step 2: Create a New Kedro Project
+Create a new directory for your project:
+```bash
+mkdir minikedro
+```
+
+Navigate into your newly created project directory:
+
+```bash
+cd minikiedro
+```
+
+### Step 3: Create `pyproject.toml`
+Create a new file named `pyproject.toml` in the project directory with the following content:
+
+```toml
+[tool.kedro]
+package_name = "minikedro"
+project_name = "minikedro"
+kedro_init_version = "0.19.9"
+source_dir = "."
+```
```

Comment:
Interesting, so the Python packaging metadata (`[project]` table) is not even needed?